documenters_aggregator/spiders/pbcc.py

```diff
@@ -0,0 +1,198 @@
+                        'classification': self._parse_classification(item),
+                        'start_time': self._parse_start(item, cal_date),
+                        'end_time': None,
+                        'all_day': self._parse_all_day(item),
+                        'status': self._parse_status(item),
+                        'location': self._parse_location(item),
+                        'sources': self._parse_sources(item)
+                    }
+
+        # Add 30 days to the current date, stop if more than 180 days (~6 months) ahead
+        self.calendar_date += timedelta(days=30)
+        if (self.calendar_date - datetime.now()).days <= 180:
+            yield self._parse_next(response)
+
+    def _parse_next(self, response):
+        """
+        Get next page. You must add logic to `next_url` and
+        return a scrapy request.
+
+        The parse method updates the calendar_date property,
+        so create a query param from that.
+        """
+        next_url = '{}?myDate={}'.format(
+            self.base_url,
+            self.calendar_date.strftime('%m/%d/%Y')
+        )
+        return scrapy.Request(next_url, callback=self.parse)
+
+    def _parse_id(self, item):
+        """
+        Calulate ID. ID must be unique within the data source being scraped.
+
+        All links for PBCC appear to include eID or BID_ID, so raise an error
+        if not found rather than creating a slug.
+        """
+        event_link = item.css('a::attr(href)').extract_first()
+        if '?eID=' in event_link:
+            id_suffix = event_link.split('?eID=')[-1]
+        elif '?BID_ID=' in event_link:
+            id_suffix = event_link.split('?BID_ID=')[-1]
+        else:
+            raise ValueError('ID is required and not present in eID or BID_ID params')
+        return 'PBCC{}'.format(id_suffix)
+
+    def _parse_classification(self, item):
+        """
+        Parse or generate classification (e.g. town hall).
+
+        PBCC has relatively helpful classifications, expanding the AFB acronym.
+        """
+        classification = item.css('a b::text').extract_first().strip()
+        if classification == 'AFB':
+            return 'Advertisement for Bids'
+        else:
+            return classification
+
+    def _parse_status(self, item):
+        """
+        Parse or generate status of meeting. Can be one of:
+
+        * cancelled
+        * tentative
+        * confirmed
+        * passed
+
+        By default, return "tentative"
+        """
+        return 'tentative'
+
+    def _parse_location(self, item):
+        """
+        Parse or generate location. Url, latitutde and longitude are all
+        optional and may be more trouble than they're worth to collect.
+
+        TODO: Location details are inconsistent across event types, addresses
+        are mixed up with names and neither are reliably in the same place.
+        Could make manual exceptions for board meetings, or go to sub-page
```

Comment:
I'd like to see an exception for board meetings. If there's a way to capture at least an address for board meetings, that would be good.